flexget/utils/titles/series.py

```diff
@@ -245,98 +244,87 @@ def parse(self, data=None, field=None, quality=None):
         log.debug('data stripped: %s', data_stripped)
 
-        # allow group(s)
-        if self.allow_groups:
-            for group in self.allow_groups:
-                group = group.lower()
-                for fmt in ['[%s]', '-%s']:
-                    if fmt % group in data_stripped:
-                        log.debug('%s is from group %s', self.data, group)
-                        self.group = group
-                        data_stripped = data_stripped.replace(fmt % group, '')
-                        break
-                if self.group:
-                    break
-            else:
-                log.debug('%s is not from groups %s', self.data, self.allow_groups)
-                return  # leave invalid
-
-        # Find quality and clean from data
-        log.debug('parsing quality ->')
-        quality = qualities.Quality(data_stripped)
-        if quality:
-            # Remove quality string from data
-            log.debug('quality detected, using remaining data `%s`', quality.clean_text)
-            data_stripped = quality.clean_text
-        # Don't override passed in quality
-        if not self.quality:
-            self.quality = quality
-
         # Remove unwanted words from data for ep / id parsing
         data_stripped = self.remove_words(data_stripped, self.remove, not_in_word=True)
 
+        groups_data_stripped = data_stripped
+
         data_parts = re.split('[\W_]+', data_stripped)
 
         for part in data_parts[:]:
             if part in self.propers:
                 self.proper_count += 1
                 data_parts.remove(part)
+                groups_data_stripped = groups_data_stripped.replace(part, '')
             elif part == 'fastsub':
                 # Subtract 5 to leave room for fastsub propers before the normal release
                 self.proper_count -= 5
                 data_parts.remove(part)
+                groups_data_stripped = groups_data_stripped.replace(part, '')
             elif part in self.specials:
                 self.special = True
                 data_parts.remove(part)
+                groups_data_stripped = groups_data_stripped.replace(part, '')
 
         data_stripped = ' '.join(data_parts).strip()
 
         log.debug("data for date/ep/id parsing '%s'", data_stripped)
 
+        matched_string = ''
+        found_identifier = False
         # Try date mode before ep mode
         if self.identified_by in ['date', 'auto']:
             date_match = self.parse_date(data_stripped)
             if date_match:
                 if self.strict_name:
                     if date_match['match'].start() > 1:
-                        return
-                self.id = date_match['date']
-                self.id_groups = date_match['match'].groups()
-                self.id_type = 'date'
-                self.valid = True
-                if not (self.special and self.prefer_specials):
-                    return
+                        found_identifier = True
```

Comment:
It looks like self.valid was being used similarly to how you are using found_identifier. What is the function of tracking it separately? Also, why does it need to continue here now instead of just return?